lib/router/router.js

```diff
@@ -224,6 +224,10 @@ export default class Router extends EventEmitter {
   }
 
   async prefetch (url) {
+    // We don't add support for prefetch in the development mode.
+    // If we do that, our on-demand-entries optimization won't performs better
+    if (process.env.NODE_ENV === 'development') return
```

Comment:
@sedubois we set NODE_ENV always. See: https://github.com/zeit/next.js/pull/1111/files#diff-0f2f34c098f5954f99483c9bd61e439dR94

We didn't use 'production' check here because, we need to allow prefetching in JEST.
So, this is safe.